imcsdk/apis/server/inventory.py

```diff
@@ -11,560 +11,306 @@
-             controller.vendor,
-             "N/A",
-             "N/A",
-             "N/A",
-             controller.fw_version))
-        ctr += 1
-    _write_to_txt_file("\n")
-
-
-def _get_storage_info(handle, file_format):
-    if file_format == "txt":
-        _get_storage_info_txt(handle)
-    elif file_format == "csv":
-        _get_storage_info_csv(handle)
-
-
-def _get_tpm_info_csv(handle):
-    with open(INVENTORY_CSV_FILE, "a") as write_handle:
-        fieldnames = ['Server-Id', 'Tpm-Id', 'Vendor',
-                      'Model', 'Serial', 'Revision']
-        writer = csv.DictWriter(write_handle, fieldnames=fieldnames)
-        writer.writeheader()
-
-        tpms = handle.query_classid("EquipmentTpm")
-        ctr = 1
-        for tpm in tpms:
-            if tpm.presence in ['missing', 'empty']:
+            host_component = x[ip][comp]
+            if len(host_component) == 0:
                 continue
-            writer.writerow({
-                'Server-Id': _extract_server_id_from_dn(handle, tpm.dn),
-                'Tpm-Id': str(ctr),
-                'Vendor': tpm.vendor,
-                'Model': tpm.model,
-                'Serial': tpm.serial,
-                'Revision': tpm.tpm_revision,
-            })
-            ctr += 1
-        writer.writerow({})
-
-
-def _get_tpm_info_txt(handle):
-    _write_to_txt_file("============= NETWORK-ADAPTOR-INFO =============\n")
-    _write_to_txt_file("TPM-ID \tVENDOR \tMODEL \tSERIAL \tREVISION\n")
-
-    tpms = handle.query_classid("EquipmentTpm")
-    ctr = 1
-    for tpm in tpms:
-        if tpm.presence in ['missing', 'empty']:
-            continue
-        _write_to_txt_file(
-            "  %s \t%s \t%s \t%s \t%s\n" %
-            (str(ctr), tpm.vendor, tpm.model, tpm.serial, tpm.tpm_revision))
-        ctr += 1
-    _write_to_txt_file("\n")
-
-
-def _get_tpm_info(handle, file_format):
-    if file_format == "txt":
-        _get_tpm_info_txt(handle)
-    elif file_format == "csv":
-        _get_tpm_info_csv(handle)
-
-
-def _delete_inventory_files(file_format):
-    import os
-
-    if file_format == "txt" or file_format == "csv":
-        file_name = "INVENTORY_" + file_format.upper() + "_FILE"
-    else:
-        return
-
-    if os.path.exists(file_name):
-        os.remove(file_name)
-
-
-def get_inventory(handle, types="all", file_format="csv"):
+            for entry in host_component:
+                row_val = []
+                for key in keys:
+                    if key not in entry:
+                        continue
+                    row_val.append(entry[key])
+                row_val.insert(0, ip)
+                html += "<tr>"
+                for each in row_val:
+                    html += "<td>" + each + "</td>"
+                html += "</tr>"
+
+    html += "</table>"
+    html += "</body>"
+    html += "</html>"
+    f.write(html)
+    f.close()
+
+
+def get_inventory(handle,
+                  component="all",
```

Comment:
Should we also handle component as a list of specific components like 
`component = ['storage', 'disks']`